hpx/runtime/applier/apply_helper.hpp

```diff
@@ -80,6 +81,12 @@ namespace hpx { namespace applier { namespace detail
                 static_cast<threads::thread_stacksize>(
                     traits::action_stacksize<Action>::value));
 
+            while (!threads::threadmanager_is_at_least(state_running))
+            {
+                boost::this_thread::sleep(boost::get_system_time() +
```

Comment:
I presume that the use of a sleep function is a case of last resort and there is no other way of waiting until the runtime is ready.